r_texture_load.c

```diff
@@ -556,7 +556,7 @@ static void R_Upload32(gltexture_t* glt, unsigned *data, int width, int height,
 
 static void R_Upload8(gltexture_t* glt, byte *data, int width, int height, int mode)
 {
-	static unsigned trans[640 * 480];
+	static unsigned trans[1024 * 1024];
```

Comment:
this one works and caused no issues for me locally 👍 
If you're unsure about any of the other changes, you can reduce this to a 1 liner and should be able to support all the common textures found in popular wads today.